tf2onnx/rewriter/dropout_rewriter.py

```diff
@@ -60,17 +71,20 @@ def rewrite_dropout(g, ops):
                 [data],
                 outputs=[out_name],
                 name=op_name,
-                attr={"ratio": 1.0},
-                shapes=[g.get_shape(inputs2.input[0])],
-                dtypes=[g.get_dtype(inputs2.input[0])]
+                attr={"ratio": ratio},
+                shapes=[g.get_shape(input2.input[0])],
+                dtypes=[g.get_dtype(input2.input[0])]
             )
             g.replace_all_inputs(ops, outputs.output[0], new_node.output[0])
-            g.safe_remove_nodes(match.get_nodes())
-
-    # remove dropout if its ratio is 1.0
-    for node in g.get_nodes():
-        if node.type == "Dropout" and node.get_attr("ratio").f == 1.0:
-            g.replace_all_inputs(g.get_nodes(), node.output[0], node.input[0])
-            g.remove_node(node.name)
+            nodes_to_remove = []
+            for node in match.get_nodes():
+                if node.name != input3.name:
+                    nodes_to_remove.append(node)
+            if g.safe_to_remove_nodes(nodes_to_remove):
+                for n in nodes_to_remove:
+                    g.remove_node(n.name)
+            else:
+                logger.error("Nodes replaced by dropout node cannot be removed because intermediate results are "
+                             "referenced elsewhere in graph")
```

Comment:
in the 'else' case, is there another option to move forward and continue with model conversion? Or it's a true halting condition?

Reply:
It logs an error but does not halt conversion.  The end result is that the intermediate nodes are left in the graph, but conversion should complete successfully.

Reply:
Ok, great. Seems like it might be better to use logger.warning(), or else to just ignore any messaging to the user. An error message is usually for fatal events.